synapse/datamodel.py

```diff
@@ -265,8 +266,10 @@ def addPropDef(self, prop, **info):
 
         ptype = info.get('ptype')
         if ptype is not None:
-            self.reqDataType(ptype)
+            dtyp = self.reqDataType(ptype)
+            pdtyp = dtyp.extend(name=dtyp.name, ppath=prop, **dtyp.info)
```

Comment:
nice use of the extend() method.  specifying **dtyp.info is not needed ( extend internally sets info defaults from self.info )
Any not to go with just "prop" for the additional info?

Reply:
Oh, and plz dont use named arg syntax for non kwargs :D   ( name= )

Reply:
Continuing to think on this, would it make sense to just specify the prop as name to the new dtype?  dunno... maybe.